corehq/apps/userreports/reports/specs.py

```diff
@@ -393,6 +394,28 @@ def get_query_column_ids(self):
         return [self.column_id]
 
 
+class SumWhenColumn(ConditionalAggregationColumn):
+    type = TypeProperty("sum_when")
+    else_ = IntegerProperty()
+
+    def get_column_config(self, data_source_config, lang):
+        return ColumnConfig(columns=[
+            DatabaseColumn(
+                header=self.get_header(lang),
+                agg_column=SumWhen(
+                    whens=self.whens,
```

Comment:
Thoughts on putting some constraints on `whens`? like making sure the values are integer types or something.